src/main/kotlin/fi/testaustime/plugin_intellij/services/TestaustimeProjectService.kt

```diff
@@ -1,32 +1,45 @@
 package fi.testaustime.plugin_intellij.services
 
 import com.intellij.notification.NotificationType
+import com.intellij.openapi.Disposable
 import com.intellij.openapi.project.Project
 import fi.testaustime.plugin_intellij.TestaustimeBundle.message
+import fi.testaustime.plugin_intellij.utils.ContextInformation.getFriendlyName
 import fi.testaustime.plugin_intellij.utils.TestaustimeNotifier
 
 
-class TestaustimeProjectService(private val project: Project) {
+class TestaustimeProjectService(private val project: Project) : Disposable {
+    private val caretListener: CaretMoveListener
+
     init {
         projects.add(project);
+        caretListener = CaretMoveListener(this);
     }
 
     fun terminate() {
         projects.remove(project);
     }
 
+    fun isActive(): Boolean {
+        return caretListener.isActive()
+    }
+
     companion object {
         var projects: MutableList<Project> = ArrayList();
         fun broadcast(tokenValid: Boolean) {
             for (project in projects) {
                 if (tokenValid) {
```

Comment:
Notifications could become annoying after some time. Would it be a good idea to let the user to disable them completely in settings? 😄

Reply:
I'm talking about notice messages, no issues with error messages.

Reply:
Yes, though the notification system was present before this PR, so it should probably be updated separately. It'll need a flag in settings state and some sort of checkbox in the Testaustime settings.